src/transforms/aws_ec2_metadata.rs

```diff
@@ -274,8 +273,14 @@ impl MetadataClient {
             return Err(Ec2MetadataError::UnableToFetchToken.into());
         }
 
-        let body = res.into_body().concat2().compat().await?;
-        let token = body.into_bytes();
+        let body = res
+            .into_body()
+            .try_fold(BytesMut::new(), |mut store, bytes| async move {
+                store.extend_from_slice(&bytes);
+                Ok(store)
+            })
+            .await?;
+        let token: Bytes = body.into();
```

Comment:
We can not use `hyper13::body::to_body` because another parties use bytes@0.4?
Maybe use with extra copy?
```
        let body = hyper13::body::to_bytes(res.into_body()).await?;
        let token = Bytes::from(&body[..]);
```

Or add function `to_body` with `try_fold` inside at top-level which later can be easily changed to `hyper13::body::to_body`?

Reply:
Yes, there is a difference in version for `bytes` crate. 

Better some extra code than an extra copy of every incoming byte. 

`to_body` function sounds good. I'll add it to this file, and when we get a need for it in some other location, then we should move it, probably to a new file, or as an extension in `src/stream.rs`, will see.